tests/runtest.py

```diff
@@ -224,6 +225,9 @@ def build_libabc(self, cflags='', ldflags=''):
 
         lib_cflags = build_cflags + ' -shared -fPIC'
 
+        if cflags != '':
```

Comment:
I'm not sure if it's a proper check.  Isn't the cflags always non-empty?

Reply:
No, in some cases, the cflags is empty.
```
$ git grep build_libabc
tests/runtest.py:    def build_libabc(self, cflags='', ldflags=''):
tests/t007_library.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t073_lib_filter.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t074_lib_trigger.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t075_lib_arg.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t076_lib_replay_F.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t077_lib_replay_T.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t131_lib_dlopen.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t168_lib_nested.py:        if TestBase.build_libabc(self, '', '') != 0:
tests/t192_lib_name.py:        if TestBase.build_libabc(self, '', '') != 0:
tests/t246_report_srcline2.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t248_dynamic_dlopen.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t266_patchable_dynamic4.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t274_filter_loc1.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t275_filter_loc2.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
tests/t276_filter_loc3.py:        if TestBase.build_libabc(self, cflags, ldflags) != 0:
```
The proper check is to test whether the `cflags` contains `-pg`, `-finstrument-functions` or `-fpatchable-function-entry`, but I think it's not needed because if `cflags` is non-empty, then it will have one of the tracable compiler options.

Reply:
In my understanding it's the default argument and the test cases pass the cflags from its caller which has tracing and optimization flags.

Reply:
As I posted above, `t168_lib_nested.py` and `t192_lib_name.py` pass an empty string to `cflags` so we need to handle these cases. Without having this check, they fail as follows.
```
168 lib_nested          : OK OK OK OK OK OK OK OK OK OK NG NG NG NG NG  OK OK OK OK OK OK OK OK OK OK NG NG NG NG NG
192 lib_name            : OK OK OK OK OK OK OK OK OK OK NG NG NG NG NG  OK OK OK OK OK OK OK OK OK OK NG NG NG NG NG
```

Reply:
Yeah, but in those cases they don't want to trace functions in the library.  I think we should check the condition explicitly like

```python
if 'patchable-function-entry' in cflags:
    self.p_libs.append('libabc_test_lib.so')
```

Reply:
Ack. I changed it to `if '-fpatchable-function-entry' in cflags`.